linkerd/app/outbound/src/http/endpoint.rs

```diff
@@ -1,75 +1,95 @@
 use super::require_identity_on_endpoint::NewRequireIdentity;
 use super::Endpoint;
-use crate::tcp;
+use crate::{tcp, Outbound};
 use linkerd_app_core::{
-    classify,
-    config::ConnectConfig,
-    http_tracing, metrics,
+    classify, config, http_tracing,
     proxy::{http, tap},
-    reconnect, svc, tls, Error, CANONICAL_DST_HEADER, L5D_REQUIRE_ID,
+    reconnect, svc, Error, CANONICAL_DST_HEADER, L5D_REQUIRE_ID,
 };
 use tokio::io;
 use tracing::debug_span;
 
-pub fn stack<B, C>(
-    config: &ConnectConfig,
-    local_id: Option<&tls::LocalId>,
-    tcp_connect: C,
-    tap: tap::Registry,
-    metrics: metrics::Proxy,
-    span_sink: http_tracing::OpenCensusSink,
-) -> impl svc::NewService<
-    Endpoint,
-    Service = impl svc::Service<
-        http::Request<B>,
-        Response = http::Response<http::BoxBody>,
-        Error = Error,
-        Future = impl Send,
-    >,
-> + Clone
+impl<C> Outbound<C>
 where
-    B: http::HttpBody<Error = Error> + std::fmt::Debug + Default + Send + 'static,
-    B::Data: Send + 'static,
-    C: svc::Service<Endpoint, Error = Error> + Clone + Send + Sync + Unpin + 'static,
+    C: svc::Service<Endpoint> + Clone + Send + Sync + Unpin + 'static,
     C::Response: io::AsyncRead + io::AsyncWrite + Send + Unpin,
+    C::Error: Into<Error>,
     C::Future: Send + Unpin,
 {
-    let identity_disabled = local_id.is_none();
-    svc::stack(tcp_connect)
+    pub fn push_http_endpoint<B>(
```

Comment:
nit/TIOLI: now that the various stacks are constructed by methods whose names are things like `push_http_endpoint`, instead of free fns that are basically all named `stack`, i'm not _super_ convinced each part of the HTTP stack needs to be in a separate module --- the names are pretty clear without modules to namespace them. We could probably put them all in the `http` module.

on the other hand, maybe you'd prefer to have more, smaller files for organizing the code? definitely not a blocker regardless.